### PR TITLE
feat: add cvat migration template

### PR DESCRIPTION
**Pull request by aleksandrmelnikov**
What type of PR is this?

/kind enhancement

What this PR does:
This PR adds CVAT as a workspace template to the template builder.
- DB entries and k8s operations are carried out.
This also includes fixes to the JupyterLab migraiton

Which issue(s) this PR fixes:

closes onepanelio/core#317

Special notes for your reviewer:
Make sure you carry out the steps here first:
- https://github.com/onepanelio/manifests/pull/61

### If you want to test on your cluster directly
- Pull down this branch
- Run `make docker` to push the image to docker
- Update your cluster deployment for onepanel-core to use the commit-hash image
- Wait for changes to take affect
- Exec into your onepanel-core pod
- Run `./goose status` to verify that cvat migration is executed
Note: If you're seeing crash loops, check the onepanel-core logs. The jupyter and cvat migrations throw a fatal log if an error is encountered while executing.

#### Assuming the initial migration up ran without errors:
- Check CVAT is available as a template in workspaces
- Create 1-2 workspaces with CVAT as base
- Wait for them to start up or to appear in the UI (database, k8s entries)
- Run `goose down` inside the pod.
- You should see no errors
- The workspaces should now be flagged as "terminating" in the UI
- Database entries for workspace_templates will show archived, along with the related workflow_template

### If you are developing locally, off of code
- You need to make sure your kubeconfig context is pointing to a k8s cluster.
- Pull down core-ui; run `npm install`, `ng serve`
- Wait for web to start
- Pull down this branch for core
- Compile main.go, have it running so it's taking requests

#### Assuming the initial migration up ran without errors:
- Check CVAT is available as a template in workspaces
- Create 1-2 workspaces with CVAT as base
- Wait for them to start up or to appear in the UI (database, k8s entries)
- Compile `cmd/goose.go`, run `./goose down`
- You should see no errors
- The workspaces should now be flagged as "terminating" in the UI
- Database entries for workspace_templates will show archived, along with the related workflow_template
- Test again by running `./goose up`, creating more cvat, then `./goose down`

### CODE REVIEW

File: db/20200528140124_add_cvat_workspace_template.go [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+		return err
+	}
+
+	namespaces, err := client.ListOnepanelEnabledNamespaces()
+	if err != nil {
+		return err
+	}
+
+	for _, namespace := range namespaces {
+		if err := client.DeleteWorkspace(namespace.Name, cvatTemplateName); err != nil {
```
Comment (rushtehrani) [inline db/20200528140124_add_cvat_workspace_template.go:158]: @aleksandrmelnikov @Vafilor is this supposed to be `client.ArchiveWorkspaceTemplate`? If so, the JupyterLab one needs to be fixed too.

Also note the changes I made in the this PR: https://github.com/onepanelio/core/pull/260/commits/7b570f861ef504ae8f5ef8e0593ae9e3901a22a1

Comment (aleksandrmelnikov) [reply db/20200528140124_add_cvat_workspace_template.go:158]: Two thoughts:
- If we want to support running the "down" migration after they make DB / K8S changes (or they've done some work), then yes. I think we need to make this change.
- If they do this immediately on creation, then this is fine.

Comment (aleksandrmelnikov) [reply db/20200528140124_add_cvat_workspace_template.go]: Pushed up a possible change.

**Comment by rushtehrani** [comment]
@aleksandrmelnikov I keep getting this error when I try this in an existing cluster. I've used multiple names to make sure there is no conflict.
```
2020/06/03 13:47:01 error rpc error: code = Unknown desc = Error with insert into workspace_templates_versions.  already exists.
Exiting.
```

**Pull request merged**